### PR TITLE
Fix tests that weren't running and have grown out of sync

### PR DESCRIPTION
**Pull request by keynmol**
We observed that the tests for `_test.go` files weren't actually running.

@varungandhi-src figured out that `TestMain` implementation was actually incorrect ([docs](https://pkg.go.dev/testing), fix: e5600856b74801d69ed788105667f5e52040c414)

On top of that, tests have grown out of sync, so required to be fixed:

- Update CLI.md to make sure the tests pass
- Pass correct input directory to snapshot formatter
- Re-generate LSIF snapshots

Additionally:
- Run the tests on CI with `-v` flag to see individual tests.

### Test plan

- [x] Verify that CI still passes and ensure that tests are actually run.
  ✅ https://github.com/sourcegraph/scip/actions/runs/4860777990/jobs/8665003910?pr=148#step:4:68

### CODE REVIEW

File: docs/CLI.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ COMMANDS:
 
 GLOBAL OPTIONS:
    --help, -h     show help (default: false)
-   --version, -v  print the version (default: false)
+   --version, -v  print the version (default: false)`
```
Comment (varungandhi-src) [inline docs/CLI.md:28]: It's weird that this extra backtick is coming up here, but it seems like a bug upstream and not in our code.

Comment (keynmol) [reply docs/CLI.md:28]: This looks like my bad in copying, let me remove

Comment (keynmol) [reply docs/CLI.md]: fixed in [d6a2dd8](https://github.com/sourcegraph/scip/pull/148/commits/d6a2dd884e8323c90ca684a9caef6d775506edb2)

**Review by varungandhi-src — APPROVED** [review]
LGTM, thanks for fixing this promptly.

**Comment by keynmol** [comment]
Issue for the flaky Typescript job: https://github.com/sourcegraph/scip/issues/149

**Pull request merged**